ksp_plugin/orbit_analyser.hpp

```diff
@@ -67,12 +67,11 @@ class OrbitAnalyser {
     void ResetRecurrence();
 
    private:
-    Analysis(Instant const& first_time,
-             not_null<RotatingBody<Barycentric> const*> primary);
+    explicit Analysis(Instant const& first_time);
 
     Instant first_time_;
     Time mission_duration_;
-    not_null<RotatingBody<Barycentric> const*> primary_;
+    RotatingBody<Barycentric> const* primary_ = nullptr;
```

Comment:
Please fix the usages of this member.  In particular, `primary()` must probably return an optional and  the caller(s) need to change.